src/api/files.api.js

```diff
@@ -343,11 +344,16 @@ export function loadKmlData(kmlLayer) {
  * @param {Number} [options.timeout] How long should the call wait before timing out
  * @param {string} [options.responseType] Type of data that the server will respond with. Options
  *   are 'arraybuffer', 'document', 'json', 'text', 'stream'. Default is `json`
- * @returns {Promise<AxiosResponse<any, any>>}
+ * @returns {Promise<GeoTIFF | AxiosResponse<any, any>>}
  */
 export async function getFileFromUrl(url, options = {}) {
     const { timeout = null, responseType = null } = options
-    if (/^https?:\/\/localhost/.test(url) || isInternalUrl(url)) {
+    if (url.toLowerCase().indexOf('.tif') !== -1) {
+        // attempting to load a GeoTIFF information (we do not want to load the whole file
+        // as it would break the concept of COGTIFF partial load, and lead to out of memory if
+        // given COGTIFF is big enough)
+        return geoTIFFFromUrl(url)
+    } else if (/^https?:\/\/localhost/.test(url) || isInternalUrl(url)) {
```

Comment:
Perhaps a generic function "isLocalHostUrl" could make the code easier to read, and reusable, but it looks like we don't test it that much (we only test this here, and in kmlUtils where it's in the middle of the backend check)